### PR TITLE
3090 - As a software engineer I want update Codes API to add checkCompanyCode route

### PR DESCRIPTION
**Pull request by istRazvan9**
Add new route for check CompanyId and accessCode
Add UnitTests

### CODE REVIEW

File: ppac-service-controller/mvp/src/main/kotlin/com/levi9/ppac/service/api/service/controller/mvp/CodesController.kt [hunk 1]
```diff
@@ -51,6 +50,30 @@ class CodesController(
...
+        summary = "Check if user has company rights",
+        description = "Returns UNAUTHORIZED if the code received and codeId doesn't match"
+    )
+    @ApiResponses(
+        value = [ApiResponse(responseCode = "200", description = "OK"), ApiResponse(
+            responseCode = "401",
+            description = "Unauthorized"
+        )]
+    )
+    @GetMapping("/checkCompanyCode/{codeId}")
```
Comment (19stephan93) [inline ppac-service-controller/mvp/src/main/kotlin/com/levi9/ppac/service/api/service/controller/mvp/CodesController.kt]: what name is **codeId**? please rename to **companyId**, cause you check the combination AccessCode (from header) and companyId from path; you check to see if an access code is valid for a company given by companyId

File: ppac-service-controller/mvp/src/main/kotlin/com/levi9/ppac/service/api/service/controller/mvp/CodesController.kt [hunk 2]
```diff
@@ -51,6 +50,30 @@ class CodesController(
...
+    fun checkCompanyCode(
+        @RequestHeader("AccessCode") accessCode: Int,
+        @PathVariable codeId: UUID
```
Comment (19stephan93) [inline ppac-service-controller/mvp/src/main/kotlin/com/levi9/ppac/service/api/service/controller/mvp/CodesController.kt]: rename everywhere **codeId** with **companyId**

File: ppac-service-controller/mvp/src/main/kotlin/com/levi9/ppac/service/api/service/controller/mvp/CodesController.kt [hunk 3]
```diff
@@ -51,6 +50,30 @@ class CodesController(
...
+    ): ResponseEntity<Any> {
+        if (securityContext.accessCodeIsSet()) {
+            val isCompanyCode = codeService?.isCompanyCode(securityContext.getAccessCode(), codeId)
```
Comment (19stephan93) [inline ppac-service-controller/mvp/src/main/kotlin/com/levi9/ppac/service/api/service/controller/mvp/CodesController.kt]: replace **codeId** with **companyId**

**Review by 19stephan93 — CHANGES_REQUESTED** [review]

**Comment by 19stephan93** [comment]
@istRazvan9 as described in story, please refer to codeId as companyId https://jira.levi9.com/browse/LEVDEN-3090

**Review by 19stephan93 — APPROVED** [review]

**Review by Madalina-Carausu — APPROVED** [review]

**Pull request merged**